oncopipe/oncopipe/__init__.py

```diff
@@ -207,23 +192,79 @@ def coerce_namedlist_to_string(obj):
     else:
         dest_dir, dest_file = os.path.split(dest)
     os.makedirs(dest_dir, exist_ok=True)
-
     dest = os.path.join(dest_dir, dest_file)
+    
+    return [src, dest]
+
+def absolute_symlink(src, dest, overwrite=True): 
+    """Creates an absolute symlink from any working directory.
+
+    Parameters
+    ----------
+    src : str
+        The source file or directory path.
+    dest : str
+        The destination file path. This can also be a destination
+        directory, and the destination symlink name will be identical
+        to the source file name (unless directory).
+    overwrite : boolean
+        Whether to overwrite the destination file if it exists.
+    """
+    # Prepare source and destination file paths 
+    src, dest = prepare_symlink(src, dest)
+
+    # Handle destination file if it already exists
+    if os.path.lexists(dest) and os.path.islink(dest):
+        if os.path.realpath(src) == os.path.realpath(dest) and os.path.isabs(os.readlink(dest)):
+            return
+        elif overwrite:
+            os.remove(dest)
+    assert not os.path.exists(dest), (
+        "Symbolic link already exists but points elsewhere: \n"
+        f"    Current: {dest} -> {os.readlink(dest)} \n"
+        f"    Attempted: {dest} -> {src}"
+    )
```

Comment:
I'm trying to think of a way of eliminating this near-duplicate code. What if you moved the call to `os.path.abspath()` that comes after this (line 229) to before this block and then change the calls to `os.path.realpath()` to `os.readlink()` like you did in the error message? This would imply that the link is absolute, negating the need for the extra conditional that prevents this code block from being identical to what's in the other function. 

If this approach works, then you can wrap this block as a two-argument function that can be used in both `relative_symlink()` and `absolute_symlink()`. I can explain further if I'm not making sense.

Reply:
I think I see what you're saying and will give it a shot.